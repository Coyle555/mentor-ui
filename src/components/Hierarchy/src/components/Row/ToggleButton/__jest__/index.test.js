import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { ToggleButton } from '../index';

afterEach(cleanup);

test('No children', () => {
	const tree = renderer.create(<ToggleButton childrenCount={0} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Expanded toggle button', () => {
	const tree = renderer.create(<ToggleButton childrenCount={1} expanded={true} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Collapsed toggle button', () => {
	const tree = renderer.create(<ToggleButton childrenCount={1} expanded={false} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Toggle button with a level of 3', () => {
	const tree = renderer.create(<ToggleButton childrenCount={1} level={3} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Clicking toggle button', () => {
	const onClick = jest.fn();
	const { container } = render(<ToggleButton childrenCount={1} onClick={onClick} />);

	fireEvent.click(container.querySelector('button'));
	expect(onClick).toHaveBeenCalled();
});
