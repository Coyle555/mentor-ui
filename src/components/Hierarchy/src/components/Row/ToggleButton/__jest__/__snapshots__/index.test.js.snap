// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsed toggle button 1`] = `
<button
  className="node-expand-button"
  style={
    Object {
      "left": "22px",
    }
  }
  type="button"
>
  <i
    className="fas fa-plus"
  />
</button>
`;

exports[`Expanded toggle button 1`] = `
<button
  className="node-collapse-button"
  style={
    Object {
      "left": "22px",
    }
  }
  type="button"
>
  <i
    className="fas fa-minus"
  />
</button>
`;

exports[`No children 1`] = `null`;

exports[`Toggle button with a level of 3 1`] = `
<button
  className="node-expand-button"
  style={
    Object {
      "left": "154px",
    }
  }
  type="button"
>
  <i
    className="fas fa-plus"
  />
</button>
`;
